src/Illuminate/Support/Facades/Date.php

```diff
@@ -0,0 +1,156 @@
+ * @method static \Illuminate\Support\Carbon isMutable()
+ * @method static bool isStrictModeEnabled()
+ * @method static bool localeHasDiffOneDayWords($locale)
+ * @method static bool localeHasDiffSyntax($locale)
+ * @method static bool localeHasDiffTwoDayWords($locale)
+ * @method static bool localeHasPeriodSyntax($locale)
+ * @method static bool localeHasShortUnits($locale)
+ * @method static void macro($name, $macro)
+ * @method static \Illuminate\Support\Carbon|null make($var)
+ * @method static \Illuminate\Support\Carbon maxValue()
+ * @method static \Illuminate\Support\Carbon minValue()
+ * @method static void mixin($mixin)
+ * @method static \Illuminate\Support\Carbon now($tz = null)
+ * @method static \Illuminate\Support\Carbon parse($time = null, $tz = null)
+ * @method static string pluralUnit(string $unit)
+ * @method static void resetMonthsOverflow()
+ * @method static void resetToStringFormat()
+ * @method static void resetYearsOverflow()
+ * @method static void serializeUsing($callback)
+ * @method static \Illuminate\Support\Carbon setHumanDiffOptions($humanDiffOptions)
+ * @method static bool setLocale($locale)
+ * @method static void setMidDayAt($hour)
+ * @method static \Illuminate\Support\Carbon setTestNow($testNow = null)
+ * @method static void setToStringFormat($format)
+ * @method static void setTranslator(\Symfony\Component\Translation\TranslatorInterface $translator)
+ * @method static \Illuminate\Support\Carbon setUtf8($utf8)
+ * @method static void setWeekEndsAt($day)
+ * @method static void setWeekStartsAt($day)
+ * @method static void setWeekendDays($days)
+ * @method static bool shouldOverflowMonths()
+ * @method static bool shouldOverflowYears()
+ * @method static string singularUnit(string $unit)
+ * @method static \Illuminate\Support\Carbon today($tz = null)
+ * @method static \Illuminate\Support\Carbon tomorrow($tz = null)
+ * @method static void useMonthsOverflow($monthsOverflow = true)
+ * @method static \Illuminate\Support\Carbon useStrictMode($strictModeEnabled = true)
+ * @method static void useYearsOverflow($yearsOverflow = true)
+ * @method static \Illuminate\Support\Carbon yesterday($tz = null)
+ */
+class Date extends Facade
+{
+    /**
+     * Get the registered name of the component.
+     *
+     * @return string
+     *
+     * @throws \RuntimeException
+     */
+    protected static function getFacadeAccessor()
+    {
+        return 'date';
+    }
+
+    /**
+     * Handle dynamic, static calls to the object.
+     *
+     * @param  string  $method
+     * @param  array   $args
+     *
+     * @return mixed
+     *
+     * @throws \RuntimeException
+     */
+    public static function __callStatic($method, $args)
+    {
+        $root = null;
+
+        try {
+            $root = static::getFacadeRoot();
+        } catch (ReflectionException $exception) {
+            // We mute reflection exceptions as if have Carbon as fallback if the facade root is missing.
```

Comment:
Following every other facade process, there would be one CarbonServiceProvider responsible for binding the accessor into the container, then we would never have a ReflectionException here.

Reply:
@deleugpn Good idea. I'm just wondering the impact of the provider dependency. First with this approach, if you use only some packages (illuminate/database only for example), then you have to run the `swap` manually to get dates properly created.

Then, right now, the service is optional (only handle default locale changes) and people may have chosen to disable the auto-discovery. These users will also have to call the `swap` before internal dates start to work.

Reply:
That's not true, facades works with only illuminate/support dependency, and the same dependency is needed for Carbon. But I see what you mean and will think of it.